### ISSUE TITLE
Fix S2094 FP: Documentation using the DefaultDocumentation package

### ISSUE THREAD

**Issue by denis-troller** [SonarSource/sonar-dotnet#9062]
### Description

Rule S2094 should not raise when used to document assembly and namespaces with package [DefaultDocumentation](https://github.com/Doraku/DefaultDocumentation)

### Repro steps

According to the packages documentation, you can document your assembly like this:
```
namespace YourAssemblyName
{
    /// <summary>
    /// your assembly documentation, used on the assembly page
    /// </summary>
    internal static class AssemblyDoc { } // internal so it is not visible outside the assembly
}
```

And you can document your namespace like this:
```
namespace YourNamespace
{
    /// <summary>
    /// your namespace documentation
    /// </summary>
    internal static class NamespaceDoc { } // internal so it is not visible outside the assembly
}
```

### Expected behavior

I expect rule S2094 not to raise in such cases..

### Actual behavior

Rule S2094 raises an issue.

**Comment by Tim-Pohlmann** [comment]
@denis-troller wouldn't it be easier to just ignore classes called `AssemblyDoc` and `NamespaceDoc`?

**Comment by Tim-Pohlmann** [comment]
NVM I misread the description.
